src/main/java/org/commcare/formplayer/services/RestoreFactory.java

```diff
@@ -436,7 +431,7 @@ public InputStream getRestoreXml(boolean skipFixtures) {
 
     public HttpHeaders getRequestHeaders(URI url) {
         HttpHeaders headers;
-        if (getHqAuth() == null) {
+        if (RequestUtils.requestAuthedWithHmac()) {
```

Comment:
decided this was clearer since if HMAC auth was used for the request we can't be sure that the HQ session auth is valid, even it was present.